src/__tests__/posthog-core.js

```diff
@@ -677,24 +678,128 @@ describe('_loaded()', () => {
+            )
+            expect(window.performance.getEntriesByType).toHaveBeenCalledTimes(3)
+            expect(window.performance.getEntriesByType).toHaveBeenNthCalledWith(1, 'navigation')
+            expect(window.performance.getEntriesByType).toHaveBeenNthCalledWith(2, 'paint')
+            expect(window.performance.getEntriesByType).toHaveBeenNthCalledWith(3, 'resource')
+        })
+
+        it('safely attempts to capture pageview with performance when enabled but not available in browser', () => {
+            const performance = { ...window.performance }
+            delete window.performance
+
+            try {
+                given('config', () => ({
+                    capture_pageview: true,
+                    capture_performance: true,
+                }))
+
+                given.subject()
+
+                expect(given.overrides.capture).toHaveBeenCalledWith(
+                    '$pageview',
+                    { performance: { navigation: [], paint: [], resource: [] } },
+                    { send_instantly: true }
+                )
+                expect(performance.getEntriesByType).not.toHaveBeenCalled()
+            } finally {
+                window.performance = performance
+            }
+        })
+
+        it('safely attempts to capture pageview with performance when enabled but getEntriesByType is not available in browser', () => {
+            const performance = { ...window.performance }
+            delete window.performance.getEntriesByType
+
+            try {
+                given('config', () => ({
+                    capture_pageview: true,
+                    capture_performance: true,
+                }))
+
+                given.subject()
+
+                expect(given.overrides.capture).toHaveBeenCalledWith(
+                    '$pageview',
+                    { performance: { navigation: [], paint: [], resource: [] } },
+                    { send_instantly: true }
+                )
+                expect(performance.getEntriesByType).not.toHaveBeenCalled()
+            } finally {
+                window.performance = performance
+            }
+        })
+
+        it('safely attempts to capture performance if a type of entry is not available in a browser', () => {
+            // e.g. IE does not implement performance paint timing
+            // https://developer.mozilla.org/en-US/docs/Web/API/PerformancePaintTiming
+            // even though it implements getEntriesByType
+
+            const performance = { ...window.performance }
+            delete window.performance.getEntriesByType
+
+            window.performance.getEntriesByType = jest.fn().mockImplementation((type) => {
+                if (type === 'paint') {
+                    throw new Error('IE does not implement this')
+                } else {
+                    return []
+                }
+            })
+
+            try {
+                given('config', () => ({
```

Comment:
Tests should follow the structure setup -> action -> test

Within this block config does not belong in a try-catch. Given this config is also repeated for most of your tests, set this config at the describe level and only override it at the single disabled test.